response_operations_ui/views/messages.py

```diff
@@ -1,16 +1,93 @@
+import json
 import logging
+from urllib.parse import parse_qs
 
-from flask import Blueprint, render_template, request
+from flask import Blueprint, flash, g, render_template, request, redirect, url_for
 from flask_login import login_required
 from structlog import wrap_logger
 
 from response_operations_ui.controllers import message_controllers
-from response_operations_ui.exceptions.exceptions import NoMessagesError
+from response_operations_ui.exceptions.exceptions import ApiError, NoMessagesError
+from response_operations_ui.forms import SecureMessageForm
 
 logger = wrap_logger(logging.getLogger(__name__))
+messages_bp = Blueprint('messages_bp', __name__,
+                        static_folder='static', template_folder='templates')
 
-messages_bp = Blueprint("messages_bp", __name__, static_folder='Static',
-                        template_folder='templates')
+
+@messages_bp.route('/create-message', methods=['GET', 'POST'])
+@login_required
+def create_message():
+    form = SecureMessageForm(request.form)
+    breadcrumbs = _build_create_message_breadcrumbs()
+
+    if form.validate_on_submit():
+        # Hard coded id's until we can fetch them from UAA or passed by Reporting Units page
+        message_json = json.dumps({
+            'msg_from': "BRES",
+            'msg_to': ["f62dfda8-73b0-4e0e-97cf-1b06327a6712"],
+            'subject': form.subject.data,
+            'body': form.body.data,
+            'thread_id': "",
+            'collection_case': "CC_PLACEHOLDER",
+            'survey': form.hidden_survey.data,
+            'ru_id': "c614e64e-d981-4eba-b016-d9822f09a4fb"})
+
+        # Keep the message subject and body
+        g.form_subject_data = form.subject.data
+        g.form_body_data = form.body.data
+
+        try:
+            message_controllers.send_message(message_json)
+            flash("Message sent.")
+            return redirect(url_for('messages_bp.view_messages'))
+        except ApiError:
+            form = _repopulate_form_with_submitted_data(form)
+            form.errors['sending'] = ["Something went wrong: Message failed to send"]
+            return render_template('create-message.html',
+                                   form=form,
+                                   breadcrumbs=breadcrumbs)
+
+    else:
+        if not form.is_submitted():
+            ru_dict = parse_qs(request.args.get('ru_details'))
+            form = _populate_hidden_form_fields_from_url_params(form, ru_dict)
+
+        form.survey.text = form.hidden_survey.data
+        form.ru_ref.text = form.hidden_ru_ref.data
+        form.business.text = form.hidden_business.data
+        form.to.text = form.hidden_to.data
+
+        return render_template('create-message.html',
+                               form=form,
+                               breadcrumbs=breadcrumbs)
+
+
+def _build_create_message_breadcrumbs():
```

Comment:
I'd argue these breadcrumbs shouldn't be hardcoded and this is a step in the right way, but for now would it be worth not having it as a function just to keep it consistent with the rest of the repo?

Reply:
I disagree, before putting this (and other things) in their own function the create_message function was super long and hard to read.   This keeps the core of that function small and the functions name is highly descriptive so I see no reason to change it.

Reply:
We can add a TBL to update this across the other endpoints. We want the core view code to be as human readable as possible